v6.0 Changelog (#5053)
mattermost/docs: source/install/self-managed-changelog.md

Also see [changelog in progress](https://bit.ly/2nK3cVf) for the next release.

Lastest Mattermost Releases:
- [Release v6.0 - Feature Release](#release-v6.0-feature-release)
- [Release v5.39 - Quality Release](#release-v5-39-quality-release)
- [Release v5.38 - Feature Release](#release-v5-38-feature-release)
- [Release v5.37 - Extended Support Release](#release-v5-37-extended-support-release)
- [Release v5.36 - Feature Release](#release-v5-36-feature-release)
- [Release v5.35 - Feature Release](#release-v5-35-feature-release)

## Release v6.0 - [Feature Release](https://docs.mattermost.com/administration/release-definitions.html#feature-release)

**Release Day: 2021-10-13**

### Deprecations

> 1. [Legacy Command Line Tools](https://docs.mattermost.com/manage/command-line-tools.html). All commands have been fully replaced by [mmctl](https://docs.mattermost.com/manage/mmctl-command-line-tool.html) and new commands have been added over the last few months, making this tool a full and robust replacement. 

Reviewer: ```suggestion
1. [Legacy Command Line Tools](https://docs.mattermost.com/manage/command-line-tools.html). Most all commands have been replaced by [mmctl](https://docs.mattermost.com/manage/mmctl-command-line-tool.html) and new commands have been added over the last few months, making this tool a robust replacement. 
```

Reviewer: There are a few commands in the legacy CLI that need to be replaced, so a full deprecation was not possible in 6.0.  I still wish to communicate the deprecation as the legacy commands will be completed in mmctl in the next couple releases.

2. [Slack Import via the web app](https://docs.mattermost.com/onboard/migrating-to-mattermost.html#migrating-from-slack-using-the-mattermost-web-app). The Slack import tool accessible via the Team Setting menu has been replaced by the [mmetl](https://docs.mattermost.com/onboard/migrating-to-mattermost.html#migrating-from-slack-using-the-mattermost-mmetl-tool-and-bulk-import) tool that is much more comprehensive for the types of data it can assist in uploading. 

> 3. MySQL versions below 5.7.7. Minimum support is now for 5.7.12. This version introduced a native JSON data type that lets us improve performance and scalability of several database fields (most notably Users and Posts props). Additionally, version 5.6 (our current minimum version) reached [EOL in February 2021](https://www.mysql.com/support/eol-notice.html).

Reviewer 2: Technically, the JSON data type was introduced earlier. But we chose 5.7.12 because of a bug we discovered in earlier versions and maximum supported versions in AWS cloud DBs were 5.7.12 I believe. @streamer45 can correct me if I am wrong.

Reviewer 3: Should we clarify if we support 5.7.7+ or 5.7.12+?

Reviewer 4: `JSON`/`jsonb` data type in DB columns was introduced first in 6.0 . Originally MySQL support was decided for version `5.7.7` since that's the version that introduced JSON as supported datatype. However, we later discovered it had a bug (fixed in `5.7.9`). At that point we decided to support as minimum version something that is commonly used in production systems and settled for `5.7.12` since it's the one used by Amazon Aurora. We should update this to say that we officially support `5.7.12+`.

   2. Medium effort, medium downtime - This process will require SQL queries to be executed manually on the server. To avoid causing a table lock, a customer can choose to use the pt-online-schema-change tool for MySQL. For Postgres, the table locking is very minimal. The advantage is that since there are a lot of queries, the customer can take their own time to run individual queries during off-hours. All queries except #11 are safe to be executed this way. Then the usual method of (1) can be followed.
   3. High effort, low downtime - This process requires everything of (2), plus it tries to minimize the impact of query #11. We can do this by following step 2, and then starting v6 along with a running v5 server, and then monitor the application logs. As soon as the v6 application starts up, we need to bring down a v5 node. This is a delicate dance, but minimizes the downtime to only a few seconds.

>  - Focalboard v0.9.1 (released with Mattermost v6.0) requires Mattermost v5.38+.

Reviewer 5: @amyblais, this has since been updated to:
"Focalboard plugin has been renamed to Mattermost Boards, and v0.9.1 (released with Mattermost v6.0) requires Mattermost v6.0+."

>    3. High effort, low downtime - This process requires everything of (2), plus it tries to minimize the impact of query #11. We can do this by following step 2, and then starting v6 along with a running v5 server, and then monitor the application logs. As soon as the v6 application starts up, we need to bring down a v5 node. This is a delicate dance, but minimizes the downtime to only a few seconds.

Reviewer 3: Propose avoiding English phrases that may be hard to understand by our international audience, e.g. `delicate dance`

>  - Focalboard plugin has been renamed to Mattermost Boards, and v0.9.1 (released with Mattermost v6.0) requires Mattermost v6.0+.

Reviewer 5: I realized we should also add a note that Focalboard is now enabled by default:
> Focalboard plugin has been renamed to Mattermost Boards, and v0.9.1 (released with Mattermost v6.0) is now enabled by default.

 - The advanced logging configuration schema changed. This is a breaking change relative to 5.x. See updated [documentation](https://docs.mattermost.com/comply/audit-log.html).

**IMPORTANT:** If you upgrade from a release earlier than v5.39, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Highlights

> 

Reviewer 3: We're missing a highlight that Mattermost now ships as one platform with three products - Channels, Playbooks and Boards.

#### Global Product Launcher

>  - Added a global header for product navigation for Channels, Playbooks, and Boards. This is disabled on the mobile web view.

Reviewer 3: and disabled on the mobile apps?

#### Branding Changes
 - Added a new default brand theme "Denim".
 - The existing theme names and colors, including Mattermost, Organization, Mattermost Dark, and Windows Dark have been updated to the new Sapphire, Quartz, Indigo, & Onyx theme names and colours, respectively. Anyone using the existing themes will see slightly modified theme colors after their server or workspace is upgraded. The theme variables for the existing Mattermost, Organization, Mattermost Dark, and Windows Dark themes will still be accessible in [our documentation](https://docs.mattermost.com/messaging/customizing-theme-colors.html#custom-theme-examples), so a custom theme can be created with these theme variables if desired. Custom themes are unaffected by this change.
 - Updated email templates to the new branding.

#### Packaging Changes

>  - Updated in-product strings referencing E10 & E20 to new packaging.

Reviewer 3: ```suggestion
 - Updated in-product strings referencing E10 & E20 to [new packaging](https://mattermost.com/pricing).
```

>  - SAML has been moved from the highest level tier (Enterprise - former E20) to mid-level (Professional) while Certificate-based authentication feature had no change.

Reviewer 3: We should mention all packaging changes listed at https://docs.google.com/document/d/1nDO4NZwwxSNm_h-meJXitnNajETQ2rPam1X4D6bwfDo/edit#bookmark=id.5rz0tzrw7n8w

Similar features can be combined, e.g. `SSO with OpenID Connect, SAML 2.0, Google and O365 moved from ....`

 - E20, Professional, and Enterprise license SKUs are now supported for installing Enterprise plugins.

#### Beta features Promoted to General Availability
   - Archived channels
   - Compliance exports
   - Custom terms of service

>    - Guest Accounts

Reviewer 3: Propose consistent capitalization, e.g. `Guest accounts`

 - Added a tutorial tip for new settings and status buttons.
 - Added a tip for the product switcher. This tip is skipped if not applicable.

### Improvements

#### User Interface (UI)
 - Added “Invite People” to the main "+" button below the hamburger menu.
 - The whole category bounds are now highlighted while holding a channel above a category name on the left-hand side.
 - Updated **Account Settings > Display > Timezone** to be more user friendly.
 - New theme agnostic file preview modal takes up the full screen. The file preview now has information about the user, channel, and the file, and moves away from text-based buttons to icon-based buttons.
 - Increased the limit of uploaded file attachments per post from 5 to 10.
 - Added desktop notifications for followed Threads.
 - Hungarian and English-Australian are now official languages.
 - Added a **Start Trial** call-to action to the **Main Menu**.
 - Changed H1-H3 heading font from Open Sans to Metropolis.

#### Performance
 - Improved typing performance when the emoji autocomplete is open.

#### Integrations
 - Dropped support for left-hand side-specific bot icons.

>  - Add a "rest field" to the App command parser.

Reviewer 3: `Added....`

 - Added support for React components in channel header tooltips registered by plugins.
 - Exported ``ChannelInviteModal`` and ``ChannelMembersModal`` components for plugins.
 - Removed a deprecated "Backend" field from the plugin manifest.
 - Converted the "Executables" field in the plugin manifest to a map.

#### Administration

>  - Added ``playbooks`` and ``boards`` to restricted team URLs list.

Reviewer 3: Do we want to mention what happens if a team currently uses these as URLs? (could be considered a breaking change)

Author: @jwilander ?

Reviewer 6: They will generally continue to work but may break in some cases per https://docs.google.com/document/d/1Slo3ZWjy_uDIS3-vvzBkuT_7QRSQmVvEcxoYjf3g33E/edit#bookmark=id.n00o6dvxa7m2

Reviewer 3: Let's note the cases that may break in the release notes

 - Focalboard plugin has been renamed to Mattermost Boards, and v0.9.1 (released with Mattermost v6.0) is now enabled by default.
 - The advanced logging configuration schema changed. This is a breaking change relative to 5.x. See updated [documentation](https://docs.mattermost.com/comply/audit-log.html).

**IMPORTANT:** If you upgrade from a release earlier than v5.39, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Highlights

#### Multi-Product Support
 - Mattermost now ships as one platform with three products - Channels, Playbooks, and Boards.

#### Global Product Launcher
 - Added a global header for product navigation for Channels, Playbooks, and Boards. This is disabled on the mobile web view and mobile apps.

#### Branding Changes

>  - Added a new default brand theme "Denim".
>  - The existing theme names and colors, including Mattermost, Organization, Mattermost Dark, and Windows Dark have been updated to the new Sapphire, Quartz, Indigo, & Onyx theme names and colours, respectively. Anyone using the existing themes will see slightly modified theme colors after their server or workspace is upgraded. The theme variables for the existing Mattermost, Organization, Mattermost Dark, and Windows Dark themes will still be accessible in [our documentation](https://docs.mattermost.com/messaging/customizing-theme-colors.html#custom-theme-examples), so a custom theme can be created with these theme variables if desired. Custom themes are unaffected by this change.

Reviewer 7: Denim was added after the initial work to update the default themes, so I've modified the details a bit:

```suggestion
 - Added a new default brand theme named "Denim".
 - The existing theme names and colors, including "Mattermost", "Organization", "Mattermost Dark", and "Windows Dark" have been updated to the new "Denim", "Sapphire", "Indigo", & "Onyx" theme names and colours, respectively. Anyone using the existing themes will see slightly modified theme colors after their server or workspace is upgraded. The theme variables for the existing "Mattermost", "Organization", "Mattermost Dark", and "Windows Dark" themes will still be accessible in [our documentation](https://docs.mattermost.com/messaging/customizing-theme-colors.html#custom-theme-examples), so a custom theme can be created with these theme variables if desired. Custom themes are unaffected by this change.
 - Added a new light theme named "Quartz" to the default available list of themes.
```

 - Added a global header for product navigation for Channels, Playbooks, and Boards. This is disabled on the mobile web view and mobile apps.

#### Branding Changes
 - Added a new default brand theme named "Denim".
 - The existing theme names and colors, including "Mattermost", "Organization", "Mattermost Dark", and "Windows Dark" have been updated to the new "Denim", "Sapphire", "Indigo", & "Onyx" theme names and colours, respectively. Anyone using the existing themes will see slightly modified theme colors after their server or workspace is upgraded. The theme variables for the existing "Mattermost", "Organization", "Mattermost Dark", and "Windows Dark" themes will still be accessible in [our documentation](https://docs.mattermost.com/messaging/customizing-theme-colors.html#custom-theme-examples), so a custom theme can be created with these theme variables if desired. Custom themes are unaffected by this change.
 - Added a new light theme named "Quartz" to the default available list of themes.
 - Updated email templates to the new branding.

#### Packaging Changes
 - Updated in-product strings referencing E10 & E20 to [new packaging](https://mattermost.com/pricing).

>  - Features moved from mid SKU to low SKU:

Reviewer 8: Does that mean they are available in E0/TE? If so, the text should make that more clear.

Author: @jasonblais ?

Reviewer 3: Yes, both E0 and TE.

@amyblais Propose `Features moved from legacy E10 to all plans, including Team Edition`

 - Updated **Account Settings > Display > Timezone** to be more user friendly.
 - New theme agnostic file preview modal takes up the full screen. The file preview now has information about the user, channel, and the file, and moves away from text-based buttons to icon-based buttons.
 - Increased the limit of uploaded file attachments per post from 5 to 10.
 - Added desktop notifications for followed Threads.
 - Hungarian and English-Australian are now official languages.
 - Added a **Start Trial** call-to action to the **Main Menu**.
 - Changed H1-H3 heading font from Open Sans to Metropolis.

#### Performance
 - Improved typing performance when the emoji autocomplete is open.
 - Added performance improvements for draft storage with multiple tabs open.
 - Improved performance of draft loading.

#### Integrations
 - Dropped support for left-hand side-specific bot icons.
 - Added a "rest field" to the App command parser.
 - Added support for React components in channel header tooltips registered by plugins.
 - Exported ``ChannelInviteModal`` and ``ChannelMembersModal`` components for plugins.
 - Removed a deprecated "Backend" field from the plugin manifest.

>  - Converted the "Executables" field in the plugin manifest to a map.

Reviewer 8: 1., 5. and 6. are breaking changes and should be marked as such. Maybe moving them into `Plugin Changes` and changing the wording would make that more clear.

Author: Added to the "Important Upgrade Notes" section.

   - ``RestrictPublicChannelCreation``
   - ``RestrictPrivateChannelCreation``
   - ``RestrictPublicChannelDeletion``
   - ``RestrictPrivateChannelDeletion``
   - ``RestrictPrivateChannelManageMembers``
   - ``DisableLegacyMFAEndpoint``
   - ``ExperimentalTownSquareIsReadOnly``
   - ``ExperimentalHideTownSquareinLHS``
   - ``EnableXToLeaveChannelsFromLHS``
   - ``CloseUnusedDirectMessages``
   - ``ExperimentalChannelOrganization``
   - ``ExperimentalChannelSidebarOrganization``
   - ``EnableLegacySidebar``
   - The legacy MFA endpoint
   - ``utils/authorization.go`` and moved any permissions to the ``MakeDefaultRoles()`` function.

### Database Changes
 - Removed deprecated ``model.ComparePassword`` method.
 - Removed deprecated ``Context.SourcePluginId`` field.
 - Removed ``(*model.Client4).CheckUserMfa``.
 - Removed ``(*model.Client4).GetServerBusyExpires``.
 - Removed MB constant from model package.
 - Removed use of pointers to the following types: 
   - ``model.ChannelList``
   - ``model.ChannelListWithTeamData``
   - ``model.ChannelMembers``
   - ``model.Preferences``
   - ``model.ProductNotices``
 - Renamed ``plugin.Context.IpAddress`` to ``plugin.Context.IPAddress``.

>  - Renamed fields in the model package to have more idiomatic names.

Reviewer 8: These aren't Database changes. A new category `Library Changes` just for this release would be the best fit.

   - [ExperimentalTownSquareIsReadOnly](https://docs.mattermost.com/configure/configuration-settings.html#town-square-is-read-only-experimental)
   - [ExperimentalHideTownSquareinLHS](https://docs.mattermost.com/configure/configuration-settings.html#town-square-is-hidden-in-left-hand-sidebar-experimental)
   - [EnableXToLeaveChannelsFromLHS](https://docs.mattermost.com/configure/configuration-settings.html#enable-x-to-leave-channels-from-left-hand-sidebar-experimental)
   - [CloseUnusedDirectMessages](https://docs.mattermost.com/configure/configuration-settings.html#autoclose-direct-messages-in-sidebar-experimental)
   - [ExperimentalChannelOrganization](https://docs.mattermost.com/configure/configuration-settings.html#sidebar-organization)
   - [ExperimentalChannelSidebarOrganization](https://docs.mattermost.com/configure/configuration-settings.html#experimental-sidebar-features)

9. [All configuration settings previously marked as “Deprecated”](https://docs.mattermost.com/configure/configuration-settings.html#deprecated-configuration-settings).

10. Changes to ``mattermost-server/model`` for naming consistency.

### Important Upgrade Notes

>  - Longer migration times can be expected. See [this document](https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055) for more details. The field type of Data in model.ClusterMessage was changed to []byte from string. Therefore, a major thing to note is that a v6 server is incompatible to run along with a v5 server in a cluster. Customers upgrading from 5.x to 6.x cannot do a High Availability upgrade. While upgrading, it is required that no other v5 server runs when a v6 server is brought up. A v6 server will run significant database schema changes that can cause a large startup time depending on the dataset size. Zero downtime will not be possible, but depending on the efforts made during the migration process, it can be minimized to a large extent. It is recommended to start Mattermost directly and not through systemctl to avoid the server process getting killed during the migration. This can happen since the value of ``TimeoutStartSec`` in the provided systemctl service file is set to 1 hour.

Reviewer 3: Is this the best document we want to link to?

Author: Upgrade docs will link to it too.

   3. High effort, low downtime - This process requires everything of (2), plus it tries to minimize the impact of query #11. We can do this by following step 2, and then starting v6 along with a running v5 server, and then monitor the application logs. As soon as the v6 application starts up, we need to bring down a v5 node. This minimizes the downtime to only a few seconds.
 - Focalboard plugin has been renamed to Mattermost Boards, and v0.9.1 (released with Mattermost v6.0) is now enabled by default.
 - The advanced logging configuration schema changed. This is a breaking change relative to 5.x. See updated [documentation](https://docs.mattermost.com/comply/audit-log.html).
 - Some breaking changes to plugins are included:
   - Support for left-hand side-specific bot icons was dropped.
   - Removed a deprecated "Backend" field from the plugin manifest.
   - Converted the "Executables" field in the plugin manifest to a map.

**IMPORTANT:** If you upgrade from a release earlier than v5.39, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Highlights

> #### Multi-Product Support

Reviewer 3: ```suggestion
#### Multi-Product Platform
```

 - Added a new light theme named "Quartz" to the default available list of themes.
 - Updated email templates to the new branding.

#### Packaging Changes
 - Updated in-product strings referencing E10 & E20 to [new packaging](https://mattermost.com/pricing).
 - Features moved from mid SKU to low SKU:
   - System default permissions, e.g. permission to create and archive channels system-wide.
      - Specifically, “System Scheme” only in **System Console > User Management > Permissions**. 
      - Existing permissions/policies in TE/E0 for "Enable Team Creation" and "Allow Team Administrators to edit others’ posts" are properly handled. 
   - Team and Channel management pages (but without channel moderation, e.g. read-only channels).

>  - Features moved from high SKU to mid SKU:

Reviewer 3: ```suggestion
 - Features moved from legacy E20 to Professional plan:
```

   - SSO with OpenID Connect, SAML, Google and O365
   - O365 integrations including MS Teams Calling and MS Calendar
   - Jira multi-server support
   - Advanced team permissions

>    - Read-only announcement channels

Reviewer: Read-only announcement channels are really Channel Moderation.  Propose we call it Channel Moderation.

>  - Features added to high-end SKU:

Reviewer: These features are not moving.  I am not certain we need to call them out specifically as they are staying in the Enterprise plan as they were previously.

Reviewer: The features listed below to be more concise.
